packages/server/src/AccountsServer.js

```diff
@@ -302,6 +320,31 @@ export class AccountsServer {
     const res = await this.db.setProfile(userId, { ...user.profile, ...profile });
     return res;
   }
+
+  async sendVerificationEmail(userId: string, address: string): Promise<void> {
+    const user = await this.db.findUserById(userId);
+    if (!user) {
+      throw new AccountsError('User not found', { id: userId });
+    }
+    // If no address provided find the first unverified email
+    if (!address) {
+      const email = find(user.emails, (e: string) => !e.verified);
+      address = email && email.address; // eslint-disable-line no-param-reassign
+    }
+    // Make sure the address is valid
+    if (!address || !includes(user.emails.map((email: string) => email.address), address)) {
+      throw new AccountsError('No such email address for user');
+    }
+    const token = await this.db.addEmailVerificationToken(userId, address);
```

Comment:
The db should layer should probably not be responsible of creating the token as this will be the same logic for every store. The only caveat with moving that logic to here is that the token should be unique and so we need the db to allow search by email verification token to make sure the "new token" is valid.